julia-package/brainflow/src/board_shim.jl

```diff
@@ -49,12 +49,35 @@ end
     file::String = ""
 end
 
+struct BoardShim
+
+    master_board_id::Int32
+    board_id::Int32
+    input_json::String
+
+    function BoardShim(id::Integer, params::BrainFlowInputParams)
+        master_id = id
+        if id == Integer(STREAMING_BOARD) || id == Integer(PLAYBACK_FILE_BOARD)
+            try
+                master_id = parse(Int, params.other_info)
+            catch
+                throw(BrainFlowError("you need to provide master board id to other_info field of BrainFlowInputParams", Integer(INVALID_ARGUMENTS_ERROR)))
+            end
+        end
+        new(master_id, id, JSON.json(params))
+    end
+
+end
+BoardShim(id::BoardIds, params::BrainFlowInputParams) = BoardShim(Integer(id), params)
+BoardShim(id) = BoardShim(id, BrainFlowInputParams())
```

Comment:
Lets remove this overload, its very important to keep API consistent across languages.

Also for almost all boards params needed, default params work only for OYmotion and Synthetic Board, it makes sense to "teach" users that they need to provide it

Reply:
Alright